improver_tests/cli/test_init.py

```diff
@@ -350,5 +351,20 @@ def test_import_cli():
     subprocess.run([sys.executable, "-c", script], check=True)  # nosec
 
 
+def test_help_no_stderr():
+    """Test if help writes to sys.stderr."""
+    import contextlib
+    import io
+
+    buffer = io.StringIO()
+    with contextlib.redirect_stderr(buffer):
+        try:
+            run_main(["improver", "help"])
```

Comment:
This does result in the output from `improver help` appearing within the standard output, when running the unit test as `pytest -vs ./improver_tests/cli/test_init.py::test_help_no_stderr`, as the `-s` option disables pytests capturing. Note that I usually use the `-s` option as it allows print statements to appear in the terminal. Would it make sense to capture the standard output within this test, in addition to the standard error, to avoid these help statements appearing when debugging?

This kind of thing appears to work:
```
    buffer = io.StringIO()
    buffer2 = io.StringIO()
    with contextlib.redirect_stderr(buffer), contextlib.redirect_stdout(buffer2):
        try:
            run_main(["improver", "help"])
        except SystemExit:
            pass
```